### PR TITLE
browser: remove non-complex uses of fs-extra for codewhisperer

### PR DESCRIPTION
**Pull request by hayemaxi**
Continuation of https://github.com/aws/aws-toolkit-vscode/pull/4230

- Adds wrapper for `copy`
- Adds function for deleting something and swalling error if it doesn't exist.
- Update tests cases where immediately relevant.
- Refactors some code from functional style to loops, so that they can be awaited.

Does not update `dependencyGraph.ts` as it contains some complex uses of `fs-extra`, such as sym
links.

#### Testing
Manual testing in progress.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/codewhisperer/service/transformByQHandler.ts [hunk 1]
```diff
@@ -402,16 +407,16 @@ export async function zipCode(modulePath: string) {
     throwIfCancelled()
 
     const tempFilePath = path.join(os.tmpdir(), 'zipped-code.zip')
-    fs.writeFileSync(tempFilePath, zip.toBuffer())
+    await fsCommon.writeFile(tempFilePath, zip.toBuffer())
```
Comment (hayemaxi) [inline src/codewhisperer/service/transformByQHandler.ts:410]: `fsCommon.writeFile` writes in `utf-8` encoding... may need to look back at some of these replacements to ensure that is correct

**Comment by justinmk3** [comment]
Needs conflict resolution

File: src/codewhisperer/service/transformByQHandler.ts (hunk 1 above)
Comment (hayemaxi) [reply src/codewhisperer/service/transformByQHandler.ts:410]: `fsCommon.writeFile` does string vs buffer distinction 👍

**Comment by justinmk3** [comment]
> `copy` in VSCode's fs api is not supported in C9

does it require a specific vscode version , and if so, which one?

**Comment by justinmk3** [comment]
Is this still relevant?

**Comment by justinmk3** [comment]
continued in https://github.com/aws/aws-toolkit-vscode/pull/5712

**Pull request closed without merging**